docs: document needed dependencies for testing components and add tests for them (#127)
asyncapi/template-for-generator-templates: README.md

This template is tested using the [Jest](https://jestjs.io/) framework. There are tests for all integral template parts, filters, hooks, and the template generation result itself. Jest-related configuration from the `package.json` file are there only because the code coverage tool conflicts with the [puppeeteer](https://github.com/puppeteer/puppeteer) library. It is possible that you do not need it in your template.

These are the contents of the `test` directory:

- `components` where you can find unit tests for the custom components
- `hooks` where you can find a snapshot test of the hook generating PNG and SVG files. In other words, a test that generates extra files, check their content and compares to the previous successful result of the test
- `helpers` where you can find unit tests for the function that generates a diagram
- `template` where you can find aa snapshot test of template generation result.

This template generates static files, so there are no examples of integration tests that would, for example, start a generated application to test if it works with a real broker.

> > NOTE: `@babel/preset-env` dependency is needed to transform your code to the appropriate version of NodeJS that you are using. `@babel/preset-react` dependency is needed to transform JSX expressions to normal JS functions. `babel` section in `package.json` apply [BabelJS](https://babeljs.io/) configuration to the `jest`.

Reviewer: ```suggestion
> NOTE:  The`@babel/preset-env` dependency is needed to transform your code to the appropriate version of NodeJS that you're using.  The `@babel/preset-react` dependency is needed to transform JSX expressions into normal JS functions.  The `babel` section in the `package.json` applies the [BabelJS](https://babeljs.io/) configuration to the `jest`.
```

Just one tiny suggestion :D
